blockchain/chaingen/generator.go

```diff
@@ -1666,18 +1668,23 @@ func (g *Generator) connectBlockTickets(b *wire.MsgBlock) {
 	}
 
 	// Extract the ticket purchases (sstx) from the block.
-	height := b.Header.Height
+	blockHash := b.BlockHash()
+	blockHeight, ok := g.blockHeights[blockHash]
```

Comment:
Rather than repeating this everywhere, please just create a function that panics.

```Go
func (g *Generator) blockHeight(hash chainhash.Hash) uint32 {
	height, ok := g.blockHeights[hash]
	if !ok {
		panic(fmt.Sprintf("no block height found for block %s", hash)
	}
}
```

Then,  you can simply do:
```Go
blockHeight := g.blockHeight(blockHash)
```

Also, notice the lowercase panic message to be consistent with everything else.